cluster-install/install.sh

```diff
@@ -1963,6 +1994,16 @@ step_collect_existing_trident_info() {
     fi
 }
 
+step_existing_trident_flags_compatibility_check() {
```

Comment:
This is an additional guardrail that was missing even when we had `SKIP_TRIDENT_UPGRADE`. Basically, if the only operations we're supposed to run are Trident-related (due to the chosen COMPONENTS), and there's already an existing Trident install, then this means the only thing we'll do is a Trident upgrade (whether it's the Trident image, operator image, ACP image, or enabling ACP).
If DO_NOT_MODIFY_EXISTING_TRIDENT=true though, then we're being explicitly told we can't do the one thing we're supposed to be doing, so we error out to let the user know of the problem.

The main reason why this is not checked earlier is because we need to collect Trident info before we know whether Trident already exists or not.